tests/preload/CMakeLists.txt

```diff
@@ -38,6 +38,12 @@ add_cstyle(tests-preload-config ${CMAKE_CURRENT_SOURCE_DIR}/config/config.c)
 add_check_whitespace(tests-preload-basic ${CMAKE_CURRENT_SOURCE_DIR}/basic/basic.c)
 add_check_whitespace(tests-preload-config ${CMAKE_CURRENT_SOURCE_DIR}/config/config.c)
 
+add_library(setumask_o OBJECT setumask.c)
```

Comment:
I think there is no need for an intermediary OBJECT target for this.
Neither is the _shared postfix needed, as we build a static version of this.
Also, I don't see why we would need to set OUTPUT_NAME.
It could just:
```
add_library(setumask SHARED setumask.c)
set_target_properties(setumask PROPERTIES INCLUDE_DIRECTORIES ${CMAKE_SOURCE_DIR}/src/)
```